client/my-sites/checkout/src/lib/pending-page.ts

```diff
@@ -189,21 +197,29 @@ export function addUrlToPendingPageRedirect(
 		urlType = 'absolute',
 		receiptId = ':receiptId',
 		fromSiteSlug,
+		fromExternalCheckout,
 	} = options;
 
 	const { origin = 'https://wordpress.com' } = typeof window !== 'undefined' ? window.location : {};
 	const successUrlPath =
 		`/checkout/thank-you/${ siteSlug || 'no-site' }/pending/` +
 		( orderId ? `${ orderId }` : ':orderId' );
-	const successUrlBase = `${ origin }${ successUrlPath }`;
+	const successUrlBase = `${
+		fromExternalCheckout ? 'https://wordpress.com' : origin
+	}${ successUrlPath }`;
 	const successUrlObject = new URL( successUrlBase );
 	successUrlObject.searchParams.set( 'redirect_to', url );
 	successUrlObject.searchParams.set( 'receiptId', String( receiptId ) );
 	if ( fromSiteSlug ) {
 		successUrlObject.searchParams.set( 'from_site_slug', fromSiteSlug );
 	}
 	if ( urlType === 'relative' ) {
-		return successUrlObject.pathname + successUrlObject.search + successUrlObject.hash;
+		return (
+			( fromExternalCheckout ? 'https://wordpress.com' : '' ) +
+			successUrlObject.pathname +
+			successUrlObject.search +
+			successUrlObject.hash
+		);
```

Comment:
For external checkouts, such as A4A client checkout, it's essential to append `https://wordpress.com` as the origin.

Reply:
Would it work to have something like this instead? I think it would be more explicitly clear that you can't have a relative URL if `fromExternalCheckout` is set.

```js
// For external checkouts, such as A4A client checkout, it's essential to append https://wordpress.com as the origin.
if ( fromExternalCheckout ) {
  urlType = 'absolute';
}
if ( urlType === 'relative' ) {
// ...
```

Or alternatively,

```js
// For external checkouts, such as A4A client checkout, it's essential to append https://wordpress.com as the origin.
if ( urlType === 'relative' && ! fromExternalCheckout ) {
//...
```

Reply:
Good point! I made some adjustments and applied your alternative suggestion. 👍